Spigot-Server-Patches/0539-Fix-anvil-not-working-when-level-req-is-0.patch

```diff
@@ -0,0 +1,51 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jan Tuck <jan@tuck.dk>
+Date: Mon, 6 Jul 2020 02:56:45 +0200
+Subject: [PATCH] Fix anvil not working when level req is 0
+
+Remove noise
+
+Fix check.
+
+diff --git a/src/main/java/net/minecraft/server/ContainerAnvil.java b/src/main/java/net/minecraft/server/ContainerAnvil.java
+index fc2038df89f25c07f6f853f6df41fe9b203c3585..c742a5a5f3ca51840d608b05b7724fef3a642719 100644
+--- a/src/main/java/net/minecraft/server/ContainerAnvil.java
++++ b/src/main/java/net/minecraft/server/ContainerAnvil.java
+@@ -38,7 +38,7 @@ public class ContainerAnvil extends ContainerAnvilAbstract {
+ 
+     @Override
+     protected boolean b(EntityHuman entityhuman, boolean flag) {
+-        return (entityhuman.abilities.canInstantlyBuild || entityhuman.expLevel >= this.levelCost.get()) && this.levelCost.get() > 0;
++        return (entityhuman.abilities.canInstantlyBuild || entityhuman.expLevel >= this.levelCost.get()) && this.levelCost.get() >= 0; //Paper
+     }
+ 
+     @Override
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index dbc46fcda3f3e006d9e7b7a89ab942b46a05cd9f..593e622814ab4d36962ea8e0bff305e3e146fab8 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -1573,6 +1573,24 @@ public class CraftEventFactory {
+         event.callEvent();
+         event.getInventory().setItem(resultSlot, event.getResult());
+         container.notifyListeners();
++
++        /*
++           I only know of this happening frequently with anvils and the level req set to 0, so i will limit it to that.
++         */
++        if (event instanceof PrepareAnvilEvent) {
++             if (((org.bukkit.inventory.AnvilInventory)view.getTopInventory()).getRepairCost() == 0) {
++                event.getViewers().forEach(viewer -> {
++                    if (viewer instanceof Player) {
++                        net.minecraft.server.EntityPlayer player = ((CraftPlayer) viewer).getHandle();
++                        boolean before = player.e; // It gotta be false for this method to work :shrug:
++                        if (player.e)
++                            player.e = false;
++                        player.broadcastCarriedItem();
++                        player.e = before;
```

Comment:
e needs obf helpers

Reply:
I used the yarn mappings name, thx proxi. Although it seems it is only used for cursor packet so no clue why it is named that, but whatever.